include/deal.II/particles/particle_handler.h

```diff
@@ -718,20 +718,81 @@ namespace Particles
     update_ghost_particles();
 
     /**
-     * Callback function that should be called before every refinement
-     * and when writing checkpoints. This function is used to
-     * register store_particles() with the triangulation. This function
-     * is used in step-70.
+     * This function prepares the particle handler for a coarsening and
+     * refinement cycle, by storing the necessary information to transfer
+     * particles to their new cells. The implementation depends on the
+     * triangulation type that is connected to the particle handler and differs
+     * between shared and distributed triangulations. This function should be
+     * used like the corresponding function with the same name in the
+     * SolutionTransfer() class.
+     */
+    void
+    prepare_for_coarsening_and_refinement();
+
+    /**
+     * This function prepares the particle handler for a coarsening and
+     * refinement cycle, by storing the necessary information to transfer
+     * particles to their new cells. The implementation depends on the
+     * triangulation type that is connected to the particle handler and differs
+     * between shared and distributed triangulations. This function should be
+     * used like the SolutionTransfer::interpolate() function after mesh
+     * refinement has finished. Note that this function requires a working
+     * mapping, i.e. if you use a mapping class that requires setup after
+     * a mesh refinement (e.g. MappingQCache(), or MappingEulerian()), the
+     * mapping has to be ready before you can call this function.
+     *
+     * @note It is important to note, that if you do not call this function
+     * after a refinement/coarsening operation (and a repartitioning operation
+     * in a distributed triangulation) and the particle handler contained
+     * particles before, the particle handler will not be usable any more. Not
+     * calling this function after refinement would be similar to trying to
+     * access a solution vector after mesh refinement without first using a
+     * SolutionTransfer class transfer the vector to the new mesh.
      */
     void
+    transfer_after_coarsening_and_refinement();
```

Comment:
For the `CellDataTransfer` class, we use the name [`unpack`](https://www.dealii.org/developer/doxygen/deal.II/classparallel_1_1distributed_1_1CellDataTransfer.html#a18bbf81e37b71e43fe827cbc45ef4101) for this function.

We now have three different functions (interpolate, unpack, transfer_after_coarsening_and_refinement) in separate classes that do essentially the same. Should we think of a common name for it?

Reply:
I agree, but they may be used in many places, so we should consider if it is worth it to rename all of them. I am not sure how many places use CellDataTransfer, but if we change the interface of SolutionTransfer, I think we have a lot of work and convincing to do :-), I would be curious about opinions though. 

Starting from a clean slate I think I would like the combination `prepare_for_mesh_transfer`/`unpack_after_mesh_transfer` best. `mesh_transfer` also captures the repartition case (currently we need to prepare for coarsening and refinement even if we only do a distributed repartition), and also references the name of the classes (CellDataTransfer/SolutionTransfer). And adding `_after_mesh_transfer` to the unpack function may not be necessary for CellDataTransfer, because its only purpose is the transfer, but I think `particle_handler.unpack()` is not quite descriptive enough, while `particle_handler.unpack_after_mesh_transfer()` make immediate sense. Also I like the symmetry ^^.

If you think those make sense, I am happy to go ahead for the particle handler. Alternatively, I am also ok with `prepare_for_coarsening_and_refinement`/`unpack` to at least make CellDataTransfer and ParticleHandler identical.